ui/src/features/project/settings/views/credentials/schema-validator.ts

```diff
@@ -0,0 +1,82 @@
+import { z } from 'zod';
+
+import { dnsRegex } from '@ui/features/common/utils';
+import { zodValidators } from '@ui/utils/validators';
+
+const helmChartRegex =
+  /^(oci:\/\/)?([a-z0-9]+(?:[._-][a-z0-9]+)*\/)*[a-z0-9]+(?:[._-][a-z0-9]+)*$/i;
+
+const imageNameRegex = /^([a-z0-9]+(?:[._-][a-z0-9]+)*\/)*[a-z0-9]+(?:[._-][a-z0-9]+)*$/i;
+
+export const createFormSchema = (genericCreds: boolean, editing?: boolean) => {
+  let schema = z
+    .object({
+      name: zodValidators.requiredString.regex(
+        dnsRegex,
+        'Credentials name must be a valid DNS subdomain.'
+      ),
+      description: z.string().optional(),
+      type: zodValidators.requiredString,
+      repoUrl: zodValidators.requiredString,
+      repoUrlIsRegex: z.boolean().optional(),
+      username: zodValidators.requiredString,
+      password: editing ? z.string().optional() : zodValidators.requiredString
+    })
+    .check(
+      z.refine(
+        (data) => {
+          if (data.type === 'git' && !data.repoUrlIsRegex) {
+            try {
+              new URL(data.repoUrl);
```

Comment:
This is closely related. Kargo supports SSH URLs, but:
1. We don't have UI support for inputting SSH URLs or PEM keys. We should.
2. I'm not sure if this new validation will suddenly start rejecting existing credentials that are SSH based.

See: https://github.com/akuity/kargo/issues/5149

Reply:
I will have this in separate PR with API support